src/vmm/tests/integration_tests.rs

```diff
@@ -288,7 +295,7 @@ fn test_dirty_bitmap_success() {
 }
 
 #[cfg(target_arch = "x86_64")]
-fn create_snapshot(is_diff: bool) {
+fn verify_create_snapshot(is_diff: bool) -> (TempFile, TempFile) {
```

Comment:
I'll take a stab at it

Reply:
I don't see an opportunity to de-duplicate any code and still keep tests clear and easy to follow.

> For example the part where we deserialize the microvm state and check if the result is the expected one can be a separate function

I personally feel that just moving those lines in two other _extra_ functions that nobody else will use, actually hurts readability with no added benefits..

Reply:
I suggested to move that part to a separate function, because in the context of verifying if the snapshot creation is successful, we shouldn't care about the specific checks (same memory size, same vcpu count) that we're doing. Those checks seem to correspond to the deserialization of a "default" microVMstate (no devices, one vcpu ...).

```rust
fn check_default_microvm_deser(memory_file: &TempFile, snapshot_file: &TempFile) {
    let restored_microvm_state: MicrovmState =
        Snapshot::load(&mut snapshot_file.as_file(), VERSION_MAP.clone()).unwrap();
   .......
    assert_eq!(restored_microvm_state.vcpu_states.len(), 1);
}
```
 I think that `verify_create_snapshot()` is more readable when you hide that logic somehow.

Moreover, I just noticed now that this part:
```rust
let boot_source_cfg: BootSourceConfig = MockBootSourceConfig::new()
                .with_default_boot_args()
                .with_kernel(NOISY_KERNEL_IMAGE)
                .into();
            let resources: VmResources = MockVmResources::new()
                .with_boot_source(boot_source_cfg)
                .into();
            let mut event_manager = EventManager::new().unwrap();
            let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();

            // The vmm will start with dirty page tracking = OFF.
            let vmm = build_microvm_for_boot(&resources, &mut event_manager, &empty_seccomp_filter)
                .unwrap();
```

 is common to a lot of tests/functions in this file. It might not be in the scope of this PR and sorry for bothering you with this :(, but the size of this file is beginning to increase substantially and it's really hard to follow the tests anymore and the scope of each one of them. Since this PR introduces two new big functions, this could be a good time to de-duplicate some code.

That function could have the following signature:
`fn default_vmm(kernel_image: Option<&str>) ->Arc<Mutex<Vmm>> { };`

If it's not the right time to do this change, feel free to dismiss my comment.

Reply:
done